Test failures detected — 2026-03-14

## Failing Tests Detected

3 tests failing in the main codebase (non-worktree). These are genuine regressions.

---

### Failure 1: Windows path separator — `packages/events/tests/decision-jsonl.test.ts`

**Test:** `createDecisionJsonlSink > uses custom baseDir`  
**Error:**
```
AssertionError: expected "vi.fn()" to be called with arguments: [ StringContaining "/custom", …(1) ]
Received: "\custom\decisions"
```

**Test:** `getDecisionFilePath > returns correct path with custom base dir`  
**Error:**
```
AssertionError: expected '\custom\decisions\run-1.jsonl' to contain '/custom'
Expected: "/custom"
Received: "\custom\decisions\run-1.jsonl"
```

**Root cause:** Test assertions use Unix-style path separators (`/custom`) but Windows produces backslash separators (`\custom`). Tests were written without cross-platform path handling.  
**Fix:** Use `path.sep`-agnostic assertions (e.g., `toContain('custom')` instead of `toContain('/custom')`, or normalize paths with `path.normalize()`).

---

### Failure 2: Invariant count drift — `packages/kernel/tests/agentguard-engine.test.ts`

**Test:** `agentguard/core/engine > createEngine > creates an engine with defaults`  
**Error:**
```
AssertionError: expected 17 to be 16 // Object.is equality
```
**Location:** `packages/kernel/tests/agentguard-engine.test.ts:18`  

**Root cause:** A new invariant was added to `BUILT_IN_INVARIANTS` (likely the env-var modification invariant from PR #392 or the container config invariant) but the hardcoded count `toBe(16)` in the test was not updated.  
**Fix:** Change `toBe(16)` to `toBe(17)`, or better, replace the magic number with `BUILT_IN_INVARIANTS.length` to prevent future drift.

---

### Infrastructure Issue: Vitest workspace includes `.claude/worktrees/` (617 false failures)

This is not a test failure per se, but it causes vitest to report 620 failing files — masking real failures. See Test Health Report #400 for full details.

**Fix needed in `vitest.workspace.ts`:** Add exclusion for `.claude/worktrees/` and `tests/*.test.js`.

---

*Detected by Test Agent (scheduled task) · See full report: #400*

--- a/packages/events/tests/decision-jsonl.test.ts
+++ b/packages/events/tests/decision-jsonl.test.ts
@@ -116,7 +116,7 @@ describe('createDecisionJsonlSink', () => {
     sink.write(makeRecord());
 
     expect(mockedMkdirSync).toHaveBeenCalledWith(
-      expect.stringContaining('/custom'),
+      expect.stringContaining('custom'),
       { recursive: true }
     );
   });
@@ -131,8 +131,8 @@ describe('getDecisionFilePath', () => {
   });
 
   it('returns correct path with custom base dir', () => {
-    const path = getDecisionFilePath('run-1', '/custom');
-    expect(path).toContain('/custom');
-    expect(path).toContain('run-1.jsonl');
+    const filePath = getDecisionFilePath('run-1', '/custom');
+    expect(filePath).toContain('custom');
+    expect(filePath).toContain('run-1.jsonl');
   });
 });

--- a/vitest.workspace.ts
+++ b/vitest.workspace.ts
@@ -3,4 +3,6 @@ import { defineWorkspace } from 'vitest/config';
 export default defineWorkspace([
   'packages/*/vitest.config.ts',
   'apps/*/vitest.config.ts',
+  '!.claude/**',
+  '!**/node_modules/**',
 ]);

